jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -697,25 +694,54 @@ YUI.add('juju-env-api', function(Y) {
     },
 
     /**
-      Define optional operations to be performed before closing the WebSocket
-      connection. Operations performed:
+      Define optional operations to be performed before logging out.
+      Operations performed:
         - the pinger interval is stopped;
+        - connection attributes are reset;
         - the mega-watcher is stopped.
       Note that not stopping the mega-watcher before disconnecting causes
       server side disconnection to take a while, therefore preventing new
       connections from being established (for instance when switching between
       models in  controller).
+      Also note that this function is intended to be idempotent: clients must
+      be free to call this multiple times even on an already closed connection.
 
-      @method beforeClose
-      @param {Function} callback A callable that must be called by the
-        function and that actually closes the connection.
+      @method cleanup
+      @param {Function} done A callable that must be called by the function and
+        that actually closes the connection.
     */
-    beforeClose: function(callback) {
+    cleanup: function(done) {
+      console.log('cleaning up the model API connection');
       if (this._pinger) {
         clearInterval(this._pinger);
         this._pinger = null;
       }
-      this._stopWatching(callback);
+      const callback = () => {
+        // TODO frankban: find a more automated way to clean up attributes.
+        this.setAttrs({
```

Comment:
I was really hoping to be able to respond with only emoji but it looks like I actually have to use words this time :laughing: 

This would be a good feature to add to my new-yet-to-have-been-written db library.